File: 01-business-sub-system/src/main/java/com/github/yingzhuo/datawarehouse/businesssubsys/dao/CartDao.scala
package com.github.yingzhuo.datawarehouse.businesssubsys.dao

import com.github.yingzhuo.datawarehouse.businesssubsys.domain.Cart
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
trait CartDao extends JpaRepository[Cart, String] {
}

File: 01-business-sub-system/src/main/java/com/github/yingzhuo/datawarehouse/businesssubsys/dao/CartDao.java
package com.github.yingzhuo.datawarehouse.businesssubsys.dao;

import com.github.yingzhuo.datawarehouse.businesssubsys.domain.Cart;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface CartDao extends JpaRepository<Cart, String> {

    public Cart findByUserId(String userId);

}
